src/blockencodings.h

```diff
@@ -109,6 +109,9 @@ class CBlockHeaderAndShortTxIDs {
     // Dummy for deserialization
     CBlockHeaderAndShortTxIDs() {}
 
+    // Unsafe init with fixed nonce for testing/internal-use only
+    CBlockHeaderAndShortTxIDs(const CBlock& block, const uint64_t nonce);
+
     CBlockHeaderAndShortTxIDs(const CBlock& block);
```

Comment:
What about always requiring a `FastRandomContext` and getting a `rand64()` for the nonce that way? peerman would pass in its own `m_rng. Making the rng a param feels preferable to adding a test-only version of the ctor.

Reply:
Sure, seems sensible to me

Reply:
Addressed in 8df9d16